metabase/metabase
# Set min bar height for non-zero bars

Closes https://github.com/metabase/metabase/issues/41291

### Description

When a bar series have a wide extent so that small values are estimated as less than 1px they can appear invisible which can be misleading. This PR sets min bar height to 1px as it was in the previous implementation.

Unfortunately, the ECharts `barMinHeight` option applies even to bars that represent zeros. To avoid that, we need to replace zeros with nulls. However, it led to not having data labels for zeros. So fixing that required hackish code both in data transformations and the labels formatting.

### How to verify

Create a bar chart
```sql
SELECT
  1 x,
  100000000 y
UNION ALL
SELECT
  2,
  100
UNION ALL
SELECT
  3,
  1
UNION ALL
SELECT
  4,
  0
UNION ALL
SELECT
  5,
  NULL
```
Enable data labels. Ensure, there is 1px bar for y = 1, no bar for y = 0, and the data label for y = 0 exists

### Demo

<img width="1720" alt="Screenshot 2024-05-31 at 10 58 33 PM" src="https://github.com/metabase/metabase/assets/14301985/1c1cc14b-74b3-459e-a77a-ff7e887eda4c">

### Checklist
